authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+        "RoleId": "ARSOMEROLEID",
+        "Arn": "arn:aws:iam::123456789012:role/openshift-machine-api-aws-cloud-credentials",
+        "CreateDate": "2021-01-06T15:54:13Z",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Federated": "$AWS_IAM_OPENID_ARN"
+                    },
+                    "Action": "sts:AssumeRoleWithWebIdentity",
+                    "Condition": {
+                        "StringEquals": {
+                            "s3.<aws-region>.amazonaws.com/$BUCKET_NAME:aud": "openshift"
+                        }
+                    }
+                }
+            ]
+        },
+        "Description": "OpenShift role for openshift-machine-api/aws-cloud-credentials",
+        "MaxSessionDuration": 3600,
+        "RoleLastUsed": {
+            "LastUsedDate": "2021-02-03T02:51:24Z",
+            "Region": "<aws-region>"
+        }
+    }
+}
+----
+
+.. To further restrict the role such that only specific cluster `ServiceAccount` objects can assume the role, update the `.Role.AssumeRolePolicyDocument.Statement[].Condition` field to the specific `ServiceAccount` object for each component.
+
+*** For `cluster-image-registry-operator` it should contain:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-image-registry:registry",
+      "system:serviceaccount:openshift-image-registry:cluster-image-registry-operator"
+    ]
+  }
+}
+----
+
+*** For `openshift-ingress-operator` it should contain:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-ingress-operator:ingress-operator"
+    ]
+  }
+}
+----
+
+*** For `openshift-cluster-csi-drivers` it should contain:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-cluster-csi-drivers:aws-ebs-csi-driver-operator",
+      "system:serviceaccount:openshift-cluster-csi-drivers:aws-ebs-csi-driver-controller-sa"
+    ]
+  }
+}
+----
+
+*** For `openshift-machine-api` it should contain:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-machine-api:machine-api-controllers"
+    ]
+  }
+}
+----
+
+. For each IAM role, attach an IAM policy to the role that reflects the required permissions. For example, for `openshift-machine-api`:
++
+[source,terminal]
+----
+$ aws iam get-role-policy --role-name openshift-machine-api-aws-cloud-credentials --policy-name openshift-machine-api-aws-cloud-credentials
```

Comment:
Does this command do the attaching or just display the policy? See: https://docs.aws.amazon.com/cli/latest/reference/iam/get-role-policy.html

Reply:
I see what you mean. @joelddiaz is this just showing them what the policy they attach should look like? And if so, should we tell them to use [attach-role-policy](https://docs.aws.amazon.com/cli/latest/reference/iam/attach-role-policy.html) and to refer to the example for what they need to put into it?

Reply:
This is an example of what the policy should look like after the policy has been created/applied..